spec/abstract/Structures_A.thy

```diff
@@ -551,10 +551,10 @@ lemma MIN_BUDGET_le_MAX_PERIOD':
 
 definition "min_sched_context_bits = 8"
 
-(* RT : size of sched_context struct in C, excluding refills
-   numbers are from MCS C: (9 * sizeof(word_t)) + (3 * sizeof(ticks_t)) for aarch32 *)
+(* RT : size of sched_context struct in C. Currently hard coded from sched_context_C_size_of.
+   See GitHub issue 501. *)
 definition sizeof_sched_context_t :: nat where
-  "sizeof_sched_context_t = (9 * 4) + (3 * 8)"
+  "sizeof_sched_context_t = 96"
 
```

Comment:
Isn't the size architecture dependent? At least I'd imagine it would be different for 32 bit and 64 bit machines. This seems to work out, i.e. `size_of(word_t)` = 8 in RSCV64, which gives you exactly the 96 instead of the 60 currently defined here (and `size_of(ticks_t)` is stable over 32/64 for now).

So I think we should keep the old comment -- the comment is correct, just not correctly implemented. It's still a hard-coded number, so linking to the github issue is also good. For the definition, we could say:

```
sizeof_sched_context_t = (9 * word_size) + (3 * 8)"
```

This should evaluate to the correct value on all architectures we currently support.

Reply:
OK great. The proofs will still go through for RISCV, but I think I'll have to revert the changes in Refine for ARM. I'll check that out